File: study/lua/src/app/scene/view/settlement/SummaryGuildReportEnd.lua
local SummaryBase = require("app.scene.view.settlement.SummaryBase")
local SummaryGuildReportEnd = class("SummaryGuildReportEnd", SummaryBase)

local ComponentBattleDesc = require("app.scene.view.settlement.ComponentBattleDesc")

function SummaryGuildReportEnd:ctor(battleData, callback, attackHurt)
    self._battleData = battleData
    local list = {}
    local size = G_ResolutionManager:getDesignCCSize()
    local width = size.width
    local height = size.height

    local midXPos = 0

    local width = math.min(1136, display.width)
    local height = math.min(640, display.height)
    -- local midXPos = 850 - width * 0.5

    local componentBattleDesc = ComponentBattleDesc.new(battleData, cc.p(midXPos, 270 - height * 0.5), ComponentBattleDesc.TYPE_REPORT)
    table.insert(list, componentBattleDesc)

    SummaryGuildReportEnd.super.ctor(self, battleData, callback, list)
end

function SummaryGuildReportEnd:onEnter()
    SummaryGuildReportEnd.super.onEnter(self)
    self:_createAnimation()
end

function SummaryGuildReportEnd:onExit()
    SummaryGuildReportEnd.super.onExit(self)
end

function SummaryGuildReportEnd:_createAnimation()
    local EffectGfxNode = require("app.effect.EffectGfxNode")
    local function effectFunction(effect)
        local subEffect = EffectGfxNode.new(effect)
        subEffect:play()
        return subEffect
    end
    G_EffectGfxMgr:createPlayMovingGfx(self, "moving_battleend", effectFunction, handler(self, self.checkStart), false)
end

return SummaryGuildReportEnd

File: study/ts/script/scene/view/settlement/SummaryGuildReportEnd.ts
import SummaryBase from "./SummaryBase";
import { G_EffectGfxMgr, G_ResolutionManager } from "../../../init";
import { handler } from "../../../utils/handler";
import ComponentBase from "./ComponentBase";
import ComponentBattleDesc from "./ComponentBattleDesc";

const { ccclass, property } = cc._decorator;
@ccclass
export default class SummaryGuildReportEnd extends SummaryBase {

    private _battleData;
    public init(battleData, callback) {
        this._battleData = battleData;
        var list: ComponentBase[] = [];

        var height = G_ResolutionManager.getDesignHeight();

        let midXPos = 0;
        var componentBattleDesc = new cc.Node("componentBattleDesc").addComponent(ComponentBattleDesc);
        componentBattleDesc.init(battleData, cc.v2(midXPos, 270 - height * 0.5), ComponentBattleDesc.TYPE_REPORT)
        list.push(componentBattleDesc);

        super.init(this._battleData, callback, list);
    }
    public onEnter() {
        super.onEnter();
        this._createAnimation();
    }
    public onExit() {
        super.onExit();
    }
    private _createAnimation() {
        G_EffectGfxMgr.createPlayMovingGfx(this.node, 'moving_battleend', null, handler(this, this.checkStart), false);
    }
}